packages/components/src/components/multi-select/multi-select/multi-select.tsx

```diff
@@ -166,40 +167,50 @@ export class MultiSelect {
 
   @Watch('value')
   public onValueChange(): void {
+    this.setFormValue(this.value);
     // When setting initial value the watcher gets called before the options are defined
     if (this.multiSelectOptions.length > 0) {
       if (!this.preventOptionUpdate) {
         setSelectedOptions(this.multiSelectOptions, this.value);
       }
       this.preventOptionUpdate = false;
-      if (this.isWithinForm) {
-        updateNativeOptions(this.nativeSelect, this.multiSelectOptions);
+    }
+  }
+
+  @Watch('form')
+  public updateFormAssociation(): void {
+    if (this.form) {
+      const formElement = document.getElementById(this.form) as HTMLFormElement;
+      if (formElement) {
+        formElement.appendChild(this.host);
       }
     }
   }
 
+  public setFormValue(value: string[]): void {
+    const formData = new FormData();
+    value.forEach((val) => formData.append(this.name, val));
+    this.internals.setFormValue(formData);
+  }
+
   public connectedCallback(): void {
     applyConstructableStylesheetStyles(this.host, getSlottedAnchorStyles);
     document.addEventListener('mousedown', this.onClickOutside, true);
-    this.form = getClosestHTMLElement(this.host, 'form');
-    this.isWithinForm = !!this.form;
     this.isNativePopoverCase = detectNativePopoverCase(this.host, false);
     if (this.isNativePopoverCase) {
       this.parentTableElement = findClosestComponent(this.host, 'pTable');
     }
   }
 
   public componentWillLoad(): void {
+    this.defaultValue = this.value;
     this.updateOptions();
     // Use initial value to set options
     setSelectedOptions(this.multiSelectOptions, this.value);
-    if (this.isWithinForm) {
-      this.nativeSelect = initNativeMultiSelect(this.host, this.name, this.disabled, this.required);
-      updateNativeOptions(this.nativeSelect, this.multiSelectOptions);
-    }
   }
 
   public componentDidLoad(): void {
+    this.setFormValue(this.value);
```

Comment:
Is it necessary to set this here? Why not in connected or willLoad

Reply:
updated